### DISCUSSION TITLE
Seeking guidance on `optimization.providedExports` et al.

### DISCUSSION THREAD

**Discussion by AprilArcus** [webpack/webpack#14148] [Q&A]
Can I get some information and guidance on these settings, which default to "true" at all times?

```
optimization.providedExports
optimization.usedExports
optimization.sideEffects
optimization.innerGraph
```

I have found that, starting from a Webpack 5 incremental rebuild time of 6.20 seconds, setting `module.unsafeCache: true` reduces my incremental rebuild time to 4.88 seconds, and that disabling these four optimization settings on top of that reduces incremental rebuild times further ​to 4.22 seconds, approaching parity with Webpack 4 at 3.68 seconds.

I do not want to disable any tree shaking functionality that would cause subtle differences between development and production mode, should a side-effect be shaken away in production but not development. Here are my questions:

* If these optimization features are left on (their default settings), can I absolutely rely on development mode and production mode having the same tree-shaking behaviors with respect to side-effects?
  * If not, what other measures must I undertake?
  * If so, can I safely disable any of these feature with respect to that intention? Which ones?

**Comment by alexander-akait** [comment]
If you disable them you will get another behavior for development mode, but if you have a pre-configured linter and check what you use only existing exports and understand how `sideEffects` work you can safely disable them for development mode
